### PR TITLE
use svd-parser Index

### LINKED ISSUE #528: Support register derivation across peripherals
The [SVD spec](https://www.keil.com/pack/doc/CMSIS/SVD/html/elem_registers.html#elem_register) supports derivedFrom directives that point to other peripherals. However, svd2rust currently just looks for the parent register in the current scope, and therefore chokes on fully-qualified names.

### PR DESCRIPTION
**Pull request by burrbull**
Should fix #528 #610 #623 #625

### CODE REVIEW

**Comment by burrbull** [comment]
cc @qwandor
Test this, please.

**Comment by burrbull** [comment]
cc @Emilgardis let's merge this

File: src/generate/peripheral.rs [hunk 1]
```diff
@@ -166,68 +156,63 @@ pub fn render(
...
+    for erc in &mut ercs {
+        match erc {
+            RegisterCluster::Cluster(c) => {
+                let mut cpath = None;
+                let dpath = c.derived_from.take();
+                if let Some(dpath) = dpath {
+                    cpath = derive_cluster(c, &dpath, &path, index)?;
+                }
+                let cpath = cpath.unwrap_or_else(|| path.new_cluster(&c.name));
+                trace!("Cluster: {}", c.name);
```
Comment (Emilgardis) [inline src/generate/peripheral.rs:187]: move the trace up

```suggestion
            RegisterCluster::Cluster(c) => {
                trace!("Cluster: {}", c.name);
                let mut cpath = None;
                let dpath = c.derived_from.take();
                if let Some(dpath) = dpath {
                    cpath = derive_cluster(c, &dpath, &path, index)?;
                }
                let cpath = cpath.unwrap_or_else(|| path.new_cluster(&c.name));
                
```

File: src/generate/peripheral.rs [hunk 2]
```diff
@@ -166,68 +156,63 @@ pub fn render(
...
-                let res: Result<TokenStream> = Err(e);
-                return handle_reg_error("Error rendering register", *reg, res);
+            RegisterCluster::Register(reg) => {
+                let mut rpath = None;
+                let dpath = reg.derived_from.take();
+                if let Some(dpath) = dpath {
+                    rpath = derive_register(reg, &dpath, &path, index)?;
+                }
+                let rpath = rpath.unwrap_or_else(|| path.new_register(&reg.name));
+                trace!("Register: {}", reg.name);
```
Comment (Emilgardis) [inline src/generate/peripheral.rs:198]: ditto

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: src/generate/peripheral.rs (hunk 1 above)
Comment (burrbull) [reply src/generate/peripheral.rs]: done

**Review by Emilgardis — APPROVED** [review]
bors r+

very nice

**Comment by qwandor** [comment]
@burrbull The resulting code builds, so that's promising. I haven't had a chance to do any more extensive tests yet.

**Pull request merged**